components/apimgt/org.wso2.carbon.apimgt.spec.parser/src/main/java/org/wso2/carbon/apimgt/spec/parser/definitions/AsyncApiParser.java

```diff
@@ -1876,258 +162,30 @@ public APIDefinitionValidationResponse validateAPIDefinition(String apiDefinitio
-                if (endpointConfig.has(APISpecParserConstants.ENDPOINT_SANDBOX_ENDPOINTS)) {
-                    AaiServer sandboxServer = getAaiServer(api, aaiDocument, endpointConfig,
-                                                        APISpecParserConstants.GATEWAY_ENV_TYPE_SANDBOX,
-                                                        APISpecParserConstants.ENDPOINT_SANDBOX_ENDPOINTS);
-                    aaiDocument.addServer(APISpecParserConstants.GATEWAY_ENV_TYPE_SANDBOX, sandboxServer);
-                }
-            }
-        }
-
-        Map<String, AaiChannelItem> channels = new HashMap<>();
-        for (URITemplate uriTemplate : api.getUriTemplates()) {
-            Aai20ChannelItem channelItem = aaiDocument.createChannelItem(uriTemplate.getUriTemplate());
-            Aai20Operation subscribeOp = new Aai20Operation(channelItem,"subscribe");
-            channelItem.subscribe = subscribeOp;
-            if (APISpecParserConstants.API_TYPE_WS.equals(api.getType())) {
-                Aai20Operation publishOp = new Aai20Operation(channelItem,"publish");
-                channelItem.publish = publishOp;
-            }
-            channels.put(uriTemplate.getUriTemplate(), channelItem);
-        }
-        aaiDocument.channels = channels;
-        return Library.writeDocumentToJSONString(aaiDocument);
-    }
 
     /**
-     * Configure Aai server from endpoint configurations
+     * Configure Async API server from endpoint configurations
      *
-     * @param api            API
-     * @param aaiDocument    AAI Document
-     * @param endpointConfig Endpoint configuration
-     * @param serverName     Name of the server
-     * @param endpoint       Endpoint to be configured
-     * @return Configured AaiServer
+     * @param api               API
+     * @param endpointConfig    Endpoint configuration
+     * @param endpoint          Endpoint to be configured
+     * @return Configured Async API Server
      */
-    private AaiServer getAaiServer(API api, Aai20Document aaiDocument, JSONObject endpointConfig, String serverName,
-                                   String endpoint) {
-
-        Aai20Server server = (Aai20Server) aaiDocument.createServer(serverName);
-        server.url = endpointConfig.getJSONObject(endpoint).getString(APISpecParserConstants.API_DATA_URL);
-        server.protocol = api.getType().toLowerCase();
-        return server;
-    }
-
-    /**
-     * Update AsyncAPI definition for store
-     *
-     * @param api            API
-     * @param asyncAPIDefinition  AsyncAPI definition
-     * @param hostsWithSchemes host addresses with protocol mapping
-     * @return AsyncAPI definition
-     * @throws APIManagementException throws if an error occurred
-     */
-    public String getAsyncApiDefinitionForStore(API api, String asyncAPIDefinition, Map<String, String> hostsWithSchemes)
+    private AsyncApiServer getAaiServer(API api, JsonObject endpointConfig, String endpoint, AsyncApiServers servers)
```

Comment:
seems you have duplicated this method across all the parser classes. Move it to utils and reuse this